### ISSUE TITLE
Possible to remove calls to cljs.compiler/munge?

### ISSUE THREAD

**Issue by arichiardi** [bhauman/lein-figwheel#508]
Hey Bruce, while working on integrating the Figwheel client in `boot` I have run across a mini-issue.

Basically it sends the js file changed event and calls `cljs.compiler/munge` on the namespace. I can replicate this no problem, but I was wondering if we could remove it for:

 - It is not part of the public API
 - It makes the `boot-reload` code depend on `clojurescript`

I checked the code and there are just three usages of it:

 - https://github.com/bhauman/lein-figwheel/blob/cc2d188ab041fc92551d3c4a8201729c47fe5846/sidecar/src/figwheel_sidecar/build_middleware/injection.clj#L176
- https://github.com/bhauman/lein-figwheel/blob/911b54e38711f689f8b54dea497bc588a902a2be/sidecar/src/figwheel_sidecar/build_middleware/notifications.clj#L26
 - https://github.com/bhauman/lein-figwheel/blob/911b54e38711f689f8b54dea497bc588a902a2be/sidecar/src/figwheel_sidecar/build_middleware/notifications.clj#L83

It seems, but please correct me if I am wrong, that we could use simple namespace keys in `notification.clj` but I am not sure of `injection.clj`.

Do you think it is doable? I can provide a patch.

Thanks a lot for you work on this!

**Comment by bhauman** [comment]
If you look at the requires in all these namespaces you will see that
figwheel does indeed depend on Clojurescript:

https://github.com/bhauman/lein-figwheel/blob/911b54e38711f689f8b54dea497bc588a902a2be/sidecar/src/figwheel_sidecar/build_middleware/notifications.clj#L6-L11

And it always will right?

And munge is used because this is how the ns names are encoded on the
client.  Google closure has a dependency tree with path information and all
the entries are the munged namespaces. The munged namespace is the main
identifier on the client and is how we reload the lib. So we can't use
namespace keys at all.

**Comment by arichiardi** [comment]
Oh ok, yes I see it depends on it and now I understand why as well, I saw the use `goog.dependencies_.` in the client. I just hoped we were not reaching for such low level/private api.

I guess figwheel will always proceed in parallel with cljs and so maybe there is no need to worry about this...closing then.

**Comment by bhauman** [comment]
BTW I know hard it is to do what you are trying to do ...  you are also one of the few people who will really understand how Figwheel works and why I haven't just simply ported it to boot. So good for you for taking this on ...

**Comment by arichiardi** [comment]
Ah ah thanks! I guess I can say the same to you 😀 I was just wondering if it makes sense to improve things for both projects, but I have another way which is to munge in `boot-cljs`. I just like the idea of having a single client that can handle browser/node/webworker and figwheel is The client at moment!
